packages/theia-integration/src/node/connection-forwarder.ts

```diff
@@ -14,35 +14,30 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 import { Channel, Disposable, DisposableCollection, MessageProvider } from '@theia/core';
-import { Message, MessageReader, MessageWriter } from 'vscode-jsonrpc';
-
-export interface IConnection extends Disposable {
-    readonly reader: MessageReader;
-    readonly writer: MessageWriter;
-    forward(to: IConnection, map?: (message: Message) => Message): void;
-    onClose(callback: () => void): Disposable;
-}
-
-// Temporary fix/workaround to enable comparability with Theia >=1.27 until https://github.com/eclipse-theia/theia/issues/11405 is resolved
-
+import { Socket } from 'net';
+import { createMessageConnection, Message, SocketMessageReader, SocketMessageWriter } from 'vscode-jsonrpc/node';
 /**
  * Forwards messages from service channel to an (raw) `vscode-json-rpc` connection
  */
 export class ConnectionForwarder implements Disposable {
     protected toDispose = new DisposableCollection();
 
-    constructor(protected readonly channel: Channel, protected readonly connection: IConnection) {
-        this.connection.reader.listen(message => this.writeMessage(message));
+    constructor(protected readonly channel: Channel, protected readonly socket: Socket) {
+        const reader = new SocketMessageReader(socket);
+        const writer = new SocketMessageWriter(socket);
+        const connection = createMessageConnection(reader, writer);
```

Comment:
If we are already touching this, maybe we can factor the create message connection out into a separate protected method that could be overridden?